src/boot/efi/boot.c

```diff
@@ -2251,28 +2251,64 @@ static void config_load_xbootldr(
 static EFI_STATUS initrd_prepare(
                 EFI_FILE *root,
                 const ConfigEntry *entry,
-                CHAR16 **ret_options) {
+                CHAR16 **ret_options,
+                UINT8 **ret_initrd,
+                UINTN *ret_initrd_size) {
 
         assert(root);
         assert(entry);
         assert(ret_options);
+        assert(ret_initrd);
+        assert(ret_initrd_size);
 
         if (entry->type != LOADER_LINUX || !entry->initrd) {
                 ret_options = NULL;
+                ret_initrd = NULL;
+                ret_initrd_size = 0;
                 return EFI_SUCCESS;
         }
 
+        /* Note that order of initrds matters. The kernel will only look for microcode updates in the very
+         * first one it sees. */
```

Comment:
not just for this, but later initrds override contents of earlier ones, if they contain same inodes